### PR TITLE
Initial Packer Integration

### PR DESCRIPTION
**Pull request by jfryman**
# Packer Integration Pack

This integration pack allows StackStorm to control [Packer](http://packer.io),
a tool for creating machine and container images for multiple platforms
from a single source configuration.

Requires `packer` to be installed on Worker nodes. See _configuration_ for
additional details.

## Configuration

* `exec_path` - full path to packer binary (default: /opt/packer/bin/packer)
* `atlas_token` - Hashicorp Atlas token, needed for `push` action.

## Actions
* `packer.build`    - Build images from a packer template
* `packer.fix`      - Takes a template and finds backwards
                      incompatible parts of it and brings it
                      up to date so it can be used with the
                      latest version of Packer
* `packer.inspect`  - Takes a template and outputs the
                      various components a template defines
* `packer.push`     - Push a template to Hashicorp Atlas
* `packer.validate` - Validate a packer template

### CODE REVIEW

**Comment by jfryman** [comment]
This is passing... I can't account for upstream module failures...

```
Command "/home/travis/build/StackStorm/st2contrib/virtualenv/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-build-vUuXDV/wit/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-NUR9TH-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/travis/build/StackStorm/st2contrib/virtualenv/include/site/python2.7/wit" failed with error code 1 in /tmp/pip-build-vUuXDV/wit
```

File: packs/packer/README.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Packer Integration Pack
```
Comment (lakshmi-kannan) [inline packs/packer/README.txt]: README.md bro.

File: packs/packer/actions/build.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+name: "build"
+runner_type: "python-script"
```
Comment (lakshmi-kannan) [inline packs/packer/actions/build.yaml]: run-python is the new thing.

File: packs/packer/actions/lib/actions.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            return variables
+
+    def set_dir(self, directory):
+        os.chdir(directory)
+
+    def packer(self, packerfile, exc=None, only=None, vars=None, vars_file=None):
+        # Cast as string from unicode to appease upstream module
+        _packerfile = str(packerfile)
+        return packer.Packer(_packerfile, exc=exc, only=only,
+                             vars=self._get_vars(vars), vars_file=vars_file,
```
Comment (lakshmi-kannan) [inline packs/packer/actions/lib/actions.py]: vars is a python keyword. use some other name perhaps?

File: packs/packer/actions/push.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from lib.actions import BaseAction
+
+
+class PushAction(BaseAction):
+    def run(self, packerfile, name, message=None, cwd=None, exclude=None,
+            only=None, variables=None, variables_file=None):
+        if cwd:
+            self.set_dir(cwd)
```
Comment (lakshmi-kannan) [inline packs/packer/actions/push.py]: Just curious, why do we need this when there is exec_path argument?

**Comment by lakshmi-kannan** [comment]
Nice PR!

File: packs/packer/actions/push.py (hunk 4 above)
Comment (jfryman) [inline packs/packer/actions/push.py]: Packer is a little fickle. It expects certain folders referenced in the definitions relative to where the script lives. To that end, you have to include this along with the definition itself.

**Pull request merged**